drivers/adc/adc_mcux_lpadc.c

```diff
@@ -200,8 +205,22 @@ static int mcux_lpadc_channel_setup(const struct device *dev,
 	}
 #endif
 
-	if (channel_cfg->reference != ADC_REF_EXTERNAL0) {
-		LOG_ERR("Invalid channel reference");
+	/*
+	 * ADC_REF_EXTERNAL1: Use SoC internal regulator as LPADC reference voltage.
+	 * ADC_REF_EXTERNAL0: Use other voltage source (maybe also within the SoCs)
+	 * as LPADC reference voltage, like VREFH, VDDA, etc.
+	 */
+	if (channel_cfg->reference == ADC_REF_EXTERNAL1) {
+		LOG_DBG("ref external1");
+		if (*regulator != NULL) {
+			regulator_set_voltage(*regulator, min_uv, max_uv);
```

Comment:
Thanks for making this update- we might want to check the return code here, and propagate it to the caller of this function if the call fails:

```
ret = regulator_set_voltage(...);
if (ret < 0) {
    return ret;
}
```